extensions/notebook/src/protocol/notebookUriHandler.ts

```diff
@@ -21,27 +21,26 @@ export class NotebookUriHandler implements vscode.UriHandler {
 	constructor() {
 	}
 
-	handleUri(uri: vscode.Uri): vscode.ProviderResult<void> {
-		switch (uri.path) {
+	handleUri(uri: vscode.Uri): Thenable<void> {
```

Comment:
I wouldn't change the return type here - you're implementing an interface and so generally should avoid modifying the return type unless necessary.

Reply:
ProviderResult also accepts Thenable, and we found that to deal with the async issues (without using done() hooks) we needed to change the return type to Thenable<void> in order to satisfy the async related issues.

Reply:
What "async issues" specifically? If you awaited the calls to handleUri that should work...was that not the case?

Reply:
This was the error from the tests, prior to changing to Thenable. 
`Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

Reply:
Hmm, ok well please push fixes for the other issues and then I'll take a look.